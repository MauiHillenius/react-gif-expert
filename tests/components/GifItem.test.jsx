import { render,screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas en <GifItem />', () => { 

    const title = 'Hello Kitty'
    const url = 'https://hello-kitty.com/hello-kitty.jpg'

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container).toMatchSnapshot();
    });

    test('Debe mostrar el url y el alt indicados ', () => {
        render(<GifItem title={ title } url={ url } />);
        const {alt, src} = screen.getByRole('img');
        expect(alt).toBe(title);
        expect(src).toBe(url);
    })

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={ title } url={ url } />);
        expect(screen.getByText( title )).toBeTruthy();
    })
 })

 