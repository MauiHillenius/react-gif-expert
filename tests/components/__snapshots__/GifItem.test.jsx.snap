// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hello Kitty"
      src="https://hello-kitty.com/hello-kitty.jpg"
    />
    <p>
      Hello Kitty
    </p>
  </div>
</div>
`;
